spirv_msl.hpp

```diff
@@ -71,6 +74,7 @@ struct MSLResourceBinding
 	spv::ExecutionModel stage = spv::ExecutionModelMax;
 	uint32_t desc_set = 0;
 	uint32_t binding = 0;
+	uint32_t size = 0;
```

Comment:
This should have a comment saying what it's for (resolving unsized arrays), so that users don't feel compelled to specify size everywhere. Could perhaps be called something else that would clarify intent as well. size would mean bit width as well for example ...

Reply:
There was already an explanation of the field in the comments. I've added an additional line clarifying that the field is not required for resources that are not held in a run-time-sized array in the shader.

I used `size` because it represents an array size. I've changed the name to `count`, which aligns with the corresponding name `descriptorCount` in Vulkan. I don't want to explicitly reference run-time-sizing in the name because having the resource count available may find other general uses later on.